# PROJECT 1 - BLOG POST

### PDX AIRBNB DATASET


### Source
Inside Airbnb is an independent, non-commercial set of tools and data that allows you to explore how Airbnb is really being used in cities around the world.
By analyzing publicly available information about a city's Airbnb's listings, Inside Airbnb provides filters and key metrics so you can see how Airbnb is being used to compete with the residential housing market.

About Inside Airbnb: http://insideairbnb.com/about.html <BR>
Dataset of Portland, OR: http://insideairbnb.com/get-the-data.html 


### Dataset files description

listings.csv.gz - Detailed Airbnb Listings data for Portland, OR.<BR>
listings.csv - Summary information and metrics for listings in Portland (good for visualisations).<BR>
reviews.csv.gz - Detailed Review Data for listings in Portland.<BR>
reviews.csv - Summary Review data and Listing ID (to facilitate time based analytics and visualisations linked to a listing).<BR>
calendar.csv.gz - Detailed Calendar Data for listings in Portland.<BR>
neighbourhoods.csv - Neighbourhood list for geo filter. Sourced from city or open source GIS files.<BR>
neighbourhoods.geojson - GeoJSON file of neighbourhoods of the city.<BR>


In [1]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_style()

Os dados disponibilizados pelo Inside Airbnb website contém 02 arquivos em formato csv com informações dos anúncios listados no site Airbnb.

- ***listings_full.csv:*** esse arquivo contém todas as informações que foram scrapped do site airbnb <br>
- ***listings.csv:*** esse arquivo é uma versão resumida do arquivo acima com apenas as colunas consieradas mais relevantes.


In [2]:
# load listings_full dataset 
listings_full = pd.read_csv('./data/listings_full.csv')
listings_full.head()

id                         listing_url       scrape_id last_scraped  \
0  12899  https://www.airbnb.com/rooms/12899  20201222151945   2020-12-22   
1  18130  https://www.airbnb.com/rooms/18130  20201222151945   2020-12-22   
2  29931  https://www.airbnb.com/rooms/29931  20201222151945   2020-12-22   
3  37676  https://www.airbnb.com/rooms/37676  20201222151945   2020-12-22   
4  41601  https://www.airbnb.com/rooms/41601  20201222151945   2020-12-22   

                                                name  \
0  Alberta Arts 2 bedroom suite, charming 1906 house   
1                          Charming Roseway Bungalow   
2       Lovely SW Victorian w/Bonus Room and Hot Tub   
3                Mt. Hood View in the Pearl District   
4                Grandpa's Bunkhouse-Backyard Studio   

                                         description  \
0  Firstly, please know that we are practicing so...   
1  An adorable 1925 bungalow home in the Roseway ...   
2  This house is wonderfully located near downtow...   
3  This 1,000 SF loft is located in the heart of ...   
4  Safety First: We have 2 studios here at our ho...   

                               neighborhood_overview  \
0  We're within walking distance of a grocery and...   
1                                                NaN   
2  While quiet and safe, the neighborhood also ha...   
3  The Pearl district enjoys a walkability score ...   
4  Ours is a neighborhood just off Noodle Flat in...   

                                         picture_url  host_id  \
0  https://a0.muscache.com/pictures/225005/cf0083...    49682   
1  https://a0.muscache.com/pictures/42c1b5ca-4aca...    69824   
2  https://a0.muscache.com/pictures/c9a3aeaf-8d99...    79786   
3  https://a0.muscache.com/pictures/212298/16fb6b...   162158   
4  https://a0.muscache.com/pictures/294ac44c-168d...   179045   

                                   host_url  ... review_scores_communication  \
0   https://www.airbnb.com/users/show/49682  ...                        10.0   
1   https://www.airbnb.com/users/show/69824  ...                        10.0   
2   https://www.airbnb.com/users/show/79786  ...                        10.0   
3  https://www.airbnb.com/users/show/162158  ...                        10.0   
4  https://www.airbnb.com/users/show/179045  ...                        10.0   

  review_scores_location review_scores_value                    license  \
0                   10.0                10.0        14-218887-000-00-HO   
1                   10.0                10.0  City registration pending   
2                   10.0                 9.0        15-142887-000-00-HO   
3                   10.0                10.0                        NaN   
4                   10.0                10.0        14-240877-000-00-HO   

  instant_bookable calculated_host_listings_count  \
0                t                              1   
1                f                              1   
2                f                              3   
3                f                              1   
4                t                              2   

  calculated_host_listings_count_entire_homes  \
0                                           1   
1                                           1   
2                                           3   
3                                           1   
4                                           2   

  calculated_host_listings_count_private_rooms  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              4.15  
1                                           0              2.00  
2                                           0              0.41  
3                             

In [4]:
# General visualization of the data information
listings_full.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3541 entries, 0 to 3540
Data columns (total 74 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            3541 non-null   int64  
 1   listing_url                                   3541 non-null   object 
 2   scrape_id                                     3541 non-null   int64  
 3   last_scraped                                  3541 non-null   object 
 4   name                                          3541 non-null   object 
 5   description                                   3533 non-null   object 
 6   neighborhood_overview                         2805 non-null   object 
 7   picture_url                                   3541 non-null   object 
 8   host_id                                       3541 non-null   int64  
 9   host_url                                      3541 non-null   o

In [68]:
listings_full.number_of_reviews.value_counts()

0      477
1      171
2      101
3       64
4       60
      ... 
472      1
464      1
462      1
458      1
629      1
Name: number_of_reviews, Length: 423, dtype: int64

**O dataset 'listings_full' contém 3541 entradas(rows) e 74 colunas.**

**Os formatos/tipos de dados das colunas(dtypes) são float64, int64 e object.**

In [28]:
listings = pd.read_csv('./data/listings.csv')
listings.head()

id                                               name  host_id  \
0  12899  Alberta Arts 2 bedroom suite, charming 1906 house    49682   
1  18130                          Charming Roseway Bungalow    69824   
2  29931       Lovely SW Victorian w/Bonus Room and Hot Tub    79786   
3  37676                Mt. Hood View in the Pearl District   162158   
4  41601                Grandpa's Bunkhouse-Backyard Studio   179045   

       host_name  neighbourhood_group neighbourhood  latitude  longitude  \
0  Ali And David                  NaN     Concordia  45.56488 -122.63418   
1          Katie                  NaN       Roseway  45.55342 -122.58002   
2            Ken                  NaN      Hayhurst  45.48278 -122.72089   
3           Paul                  NaN         Pearl  45.52555 -122.68193   
4           Jean                  NaN       Roseway  45.54935 -122.58425   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0  Entire home/apt     75               2                551  2020-10-19   
1  Entire home/apt     80               3                  2  2020-11-30   
2  Entire home/apt    200               2                 53  2019-08-31   
3  Entire home/apt    100              30                123  2019-10-11   
4  Entire home/apt    169               3                226  2020-09-05   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               4.15                               1               361  
1               2.00                               1                 0  
2               0.41                               3               204  
3               0.97                               1               365  
4               1.79                               2               365

In [29]:
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3541 entries, 0 to 3540
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              3541 non-null   int64  
 1   name                            3541 non-null   object 
 2   host_id                         3541 non-null   int64  
 3   host_name                       3539 non-null   object 
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   3541 non-null   object 
 6   latitude                        3541 non-null   float64
 7   longitude                       3541 non-null   float64
 8   room_type                       3541 non-null   object 
 9   price                           3541 non-null   int64  
 10  minimum_nights                  3541 non-null   int64  
 11  number_of_reviews               3541 non-null   int64  
 12  last_review                     30

In [72]:
num_data = listings._get_numeric_data()
num_data.columns.tolist()

['id',
 'host_id',
 'neighbourhood_group',
 'latitude',
 'longitude',
 'price',
 'minimum_nights',
 'number_of_reviews',
 'reviews_per_month',
 'calculated_host_listings_count',
 'availability_365']